src/MetricsTimer.cpp

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This code has been produced by the European Spallation Source
+// and its partner institutes under the BSD 2 Clause License.
+//
+// See LICENSE.md at the top level for license information.
+//
+// Screaming Udder!                              https://esss.se
+
+#include "MetricsTimer.h"
+#include "CURLReporter.h"
+#include "CommandHandler.h"
+#include "Converter.h"
+
+#ifdef _MSC_VER
+std::vector<char> getHostname() {
+  std::vector<char> Hostname;
+  return Hostname;
+}
+#else
+#include <unistd.h>
+std::vector<char> getHostname() {
+  std::vector<char> Hostname;
+  Hostname.resize(256);
+  gethostname(Hostname.data(), Hostname.size());
+  if (Hostname.back() != 0) {
+    // likely an error
+    Hostname.back() = 0;
+  }
+  return Hostname;
+}
+#endif
+
+namespace Forwarder {
+
+void MetricsTimer::start() {
```

Comment:
`MetricsTimer::start()` can be baked into the constructor. It is called right after the call to the destructor anyway
.

Reply:
:+1: could just make `start` a private method and call it in the constructor.